include/ddc/detail/kokkos.hpp

```diff
@@ -8,55 +8,101 @@
 
 namespace detail {
 
-template <class KokkosLP>
-struct mdspan_layout;
+template <class T>
+struct type_holder
+{
+    using type = T;
+};
 
 template <class KokkosLP>
-using mdspan_layout_t = typename mdspan_layout<KokkosLP>::type;
+struct kokkos_to_mdspan_layout
+{
+    static_assert(
```

Comment:
is this better than just not defining the struct?

Reply:
It just helps to diagnose the problem with an error message.

Reply:
:+1: